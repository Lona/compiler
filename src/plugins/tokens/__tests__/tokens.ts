import Helpers from '../../../helpers'
import { convertFile } from '../index'
import { createFs } from 'buffs'

const tokensBlock = (string: string) => '```tokens\n' + string + '\n```\n'

// TODO: Test more kinds of tokens
describe('Tokens', () => {
  it('generates tokens', async () => {
    const source = createFs({
      'lona.json': JSON.stringify({}),
      'Colors.md': tokensBlock(`let color: Color = #color(css: "pink")`),
    })

    const helpers = await Helpers(source, '/')
    const converted = await convertFile('/Colors.md', helpers)

    expect(converted).toMatchSnapshot()
  })
})
