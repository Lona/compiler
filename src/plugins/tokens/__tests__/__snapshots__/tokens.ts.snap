// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tokens generates tokens 1`] = `
Object {
  "contents": Object {
    "type": "flatTokens",
    "value": Array [
      Object {
        "qualifiedName": Array [
          "color",
        ],
        "value": Object {
          "type": "color",
          "value": Object {
            "css": "pink",
          },
        },
      },
    ],
  },
  "inputPath": "/Colors.md",
  "name": "Colors",
  "outputPath": "/Colors.flat.json",
}
`;
